src/multisketch.rs

```diff
@@ -196,6 +191,67 @@ impl MultiSketch {
 
         self
     }
+    pub fn remove_metadata(
+        &mut self,
+        output_file_name: &str,
+        genome_ids_to_remove: &[String],
+    ) -> std::io::Result<()> {
+        println!("{}", self);
+        let mut new_sketch_metadata: Vec<Sketch> = Vec::with_capacity(self.sketch_metadata.len());
+
+        for sketch in &self.sketch_metadata {
+            if !genome_ids_to_remove.contains(&(*sketch.name()).to_string()) {
+                new_sketch_metadata.push(sketch.clone());
+            }
```

Comment:
I would add an else branch here which collects any missing names, then list the missing ones and return an `Error` (not panic, see comment below).

Reply:
by missing do you mean samples that were supposed to be removed but are not in the dataset?

Reply:
exactly